iodata/formats/mol2.py

```diff
@@ -0,0 +1,188 @@
+        if len(words) == 9:
+            atchgs[i] = float(words[8])
+        else:
+            atchgs[i] = 0.0000
+    atcoords = atcoords * angstrom
+    attypes = tuple(attypes)
+    return atnums, atcoords, atchgs, attypes
+
+
+def _load_helper_bonds(lit: LineIterator, nbonds: int) -> Tuple[np.ndarray]:
+    """Load bond information."""
+    bonds = np.empty((nbonds, 3))
+    for i in range(nbonds):
+        words = next(lit).split()
+        if words[3] == 'am':
+            words[3] = 4
+        try:
+            # Substract one because of numbering starting at 0
+            bond = [int(words[1]) - 1, int(words[2]) - 1, int(words[3])]
+            bonds[i] = bond
+        except ValueError:
+            lit.error('Something wrong in the bond section')
+    return bonds
+
+
+@document_load_many("MOL2", ['atcoords', 'atnums', 'atcharges', 'atffparams'], ['title'])
+def load_many(lit: LineIterator) -> Iterator[dict]:
+    """Do not edit this docstring. It will be overwritten."""
+    # MOL2 files with more molecules are a simple concatenation of individual MOL2 files,'
+    # making it trivial to load many frames.
+    while True:
+        try:
+            yield load_one(lit)
+        except IOError:
+            return
+
+
+@document_dump_one("MOL2", ['atcoords', 'atnums'], ['atcharges', 'atffparams', 'title'])
+def dump_one(f: TextIO, data: IOData):
+    """Do not edit this docstring. It will be overwritten."""
+    # The first six lines are reserved for comments
+    print("# Mol2 file created with Iodata", file=f)
+    print("\n\n\n\n\n", file=f)
+    print("@<TRIPOS>MOLECULE", file=f)
+    print(data.title or 'Created with IOData', file=f)
+    if data.bonds is not None:
+        bonds = len(data.bonds)
+        print(f'{data.natom:5d} {bonds:6d} {0:6d} {0:6d}', file=f)
+    else:
+        print(f'{data.natom:5d} {0:6d} {0:6d} {0:6d}', file=f)
+    print("@<TRIPOS>ATOM", file=f)
+    for i in range(data.natom):
+        n = num2sym[data.atnums[i]]
+        x, y, z = data.atcoords[i] / angstrom
+        out1 = f'{i+1:7d} {n:2s} {x:15.4f} {y:9.4f} {z:9.4f} '
+        atcharges = data.atcharges.get('mol2charges')
+        attypes = data.atffparams.get('attypes')
+        if atcharges is not None and attypes is not None:
+            charge = atcharges[i]
+            attype = attypes[i]
+            out2 = f'{attype:6s} {1:4d} XXX {charge:14.4f}'
+        elif atcharges is not None:
+            charge = atcharges[i]
+            out2 = f'{n:6s} {1:4d} XXX {charge:14.4f}'
+        elif attypes is not None:
+            charge = 0.0000
+            attype = attypes[i]
+            out2 = f'{attype:6s} {1:4d} XXX {charge:14.4f}'
+        else:
+            charge = 0.0000
+            out2 = f'{n:6s} {1:4d} XXX {charge:14.4f}'
+        print(out1 + out2, file=f)
+    if data.bonds is not None:
+        print("@<TRIPOS>BOND", file=f)
+        bonds = data.bonds
+        for i, bond in enumerate(bonds):
+            if bond[2] == 4:
```

Comment:
Code repetition can be avoided, more or less along the following lines:

```python
bondtype = 'am' if bond[2] == 4 else bond[2]
print(f'{i+1:6d} {bond[0]+1:4d} {bond[1]+1:4d} {bondtype:2s}', file=f)
```

Reply:
Again, thanks. Changed accordingly.